### PR TITLE
[DG] ADY-12 template for cucumber and config

### PR DESCRIPTION
**Pull request by pandemicGander**

### CODE REVIEW

File: src/main/resources/config/application.yml [hunk 1]
```diff
@@ -106,3 +106,10 @@ jhipster:
 # ===================================================================
 
 application:
+
+miraklOperator:
```
Comment (rselimi) [inline src/main/resources/config/application.yml]: Why do you need this for dev and prod too? It's always going to be injected as a system property.

Comment (pandemicGander) [reply src/main/resources/config/application.yml]: The backing config bean needs it as far as I know

```org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'miraklFrontCredential' defined in class path resource [com/adyen/mirakl/config/MiraklFrontConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.mirakl.client.core.security.MiraklCredential]: Factory method 'miraklFrontCredential' threw exception; nested exception is java.lang.IllegalArgumentException: the apiKey cannot be null
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:599)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory
```

When the env vars exist but we do not mention anything in `application.yml`, seems we cannot create our configuration e.g. `MiraklFrontConfiguration`

Comment (rselimi) [reply src/main/resources/config/application.yml]: Not doubting but it seems non-standard to me. We should avoid repeating same values in multiple config layers if possible.

Comment (pandemicGander) [reply src/main/resources/config/application.yml]: I'll take out of `dev` and `prod` profile for now as they override `application.yml`

File: src/main/java/com/adyen/mirakl/config/MiraklProperties.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.adyen.mirakl.config;
+
+public abstract class MiraklProperties {
+
+    private String miraklEnvUrl;
+    private String miraklApiKey;
```
Comment (rmcotton) [inline src/main/java/com/adyen/mirakl/config/MiraklProperties.java]: can you call this miraklOperatorApiKey

File: src/main/resources/config/application.yml [hunk 3]
```diff
@@ -106,3 +106,10 @@ jhipster:
...
+    miraklEnvUrl: ${MIRAKL_ENV_URL}
+    miraklApiKey: ${MIRAKL_API_OPERATOR_KEY}
+miraklFront:
+    miraklEnvUrl: ${MIRAKL_ENV_URL}
```
Comment (rmcotton) [inline src/main/resources/config/application.yml]: is there anyway we can separate our test requirements vs the actual live application?

We only need the front key for testing purposes.

File: src/main/resources/config/application.yml (hunk 1 above)
Comment (rmcotton) [reply src/main/resources/config/application.yml]: note we will have different urls for different mirakl environments

i.e. customer-dev.mirakl.net  customer-preprod.mirakl.net  customer-prod.mirakl.net   or even marketplace.customer.com

File: src/main/resources/config/application.yml (hunk 3 above)
Comment (rmcotton) [reply src/main/resources/config/application.yml]: okay now i've read through the rest of the PR. please remove frontkey from the app itself.

**Pull request merged**